src/Orchard.Cms.Web/Modules/Orchard.OpenId/Services/OpenIdTokenStore.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenIddict.Core;
+using Orchard.OpenId.Indexes;
+using Orchard.OpenId.Models;
+using YesSql.Core.Services;
+
+namespace Orchard.OpenId.Services
+{
+    public class OpenIdTokenStore : IOpenIddictTokenStore<OpenIdToken>
+    {
+        private readonly ISession _session;
+
+        public OpenIdTokenStore(ISession session)
+        {
+            _session = session;
+        }
+
+        public async Task<OpenIdToken> CreateAsync(OpenIdToken token, CancellationToken cancellationToken)
+        {
+            if (token == null)
+            {
+                throw new ArgumentNullException(nameof(token));
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+            
+            _session.Save(token);
+            await _session.CommitAsync();
+
+            return token;
+        }
+
+        public Task<OpenIdToken> CreateAsync(string type, string subject, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrEmpty(type))
+            {
+                throw new ArgumentException("The token type cannot be null or empty.");
```

Comment:
No, `ArgumentNullException` is for null checks only (here we also check if the string is empty).